TestsFoundation/TestsFoundation/Extensions/TestTreeExtensions.swift

```diff
@@ -0,0 +1,117 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2020-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+@testable import Core
+
+extension TestTree: CustomStringConvertible {
+    public var description: String { description(indent: "", after: "").joined(separator: "\n") }
+    private func description(indent: String, after: String) -> [String] {
+        var result = ["\(indent)\(subtrees.isEmpty ? "" : "┬") \(id ?? "type: \(String(reflecting: type))")"]
+        if let info = info {
+            for line in String(reflecting: info).split(separator: "\n") {
+                result.append("\(after)\(subtrees.isEmpty ? "" : "│")   \(line)")
+            }
+        }
+        for subtree in subtrees.dropLast() {
+            result.append(contentsOf: subtree.description(indent: "\(after)├─", after: "\(after)│ "))
+        }
+        if let subtree = subtrees.last {
+            result.append(contentsOf: subtree.description(indent: "\(after)└─", after: "\(after)  "))
+        }
+        return result
+    }
+
+    public struct NotFoundError: Error {
+        let description: String
+        init(_ description: String = "") {
+            self.description = description
+        }
+    }
+
+    // convenience search methods
+    // findAll:  all matches, any descendant
+    // find:     first match, any descendant
+    // children: all matches, children only
+    // child:    first match, children only
```

Comment:
I couldn't figure out a more orthogonal way to define these helpers that was still easy to use